src/VisualStudio/Core/Def/Diagnostics/VisualStudioCodeAnalysisSuggestionsConfigService.cs

```diff
@@ -0,0 +1,231 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeAnalysisSuggestions;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.CodeAnalysisSuggestions;
+
+[ExportWorkspaceService(typeof(ICodeAnalysisSuggestionsConfigService), ServiceLayer.Host), Shared]
+[Export(typeof(VisualStudioCodeAnalysisSuggestionsConfigService))]
```

Comment:
why workspace service, and directly exported?

Reply:
> why workspace service

need to fetch this service from CodeRefactoringProvider

> and directly exported?

Removed now